### PR TITLE
feat!: upgrade to helia 5

### PR DESCRIPTION
**Pull request by 2color**
## Description 

- Upgrade Helia to v5 and js-libp2p to v2
- Fix breaking changes

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/verified-fetch/src/utils/get-tar-stream.ts [hunk 1]
```diff
@@ -28,7 +27,7 @@ function toHeader (file: UnixFSEntry): Partial<TarEntryHeader> & { name: string
 
 function toTarImportCandidate (entry: UnixFSEntry): TarImportCandidate {
   if (!EXPORTABLE.includes(entry.type)) {
-    throw new CodeError('Not a UnixFS node', 'ERR_NOT_UNIXFS')
+    throw new Error('Not a UnixFS node')
```
Comment (achingbrain) [inline packages/verified-fetch/src/utils/get-tar-stream.ts:30]: You might want to use named errors for better DX here and elsewhere in this PR?

`@helia/unixfs` exports a `NotUnixFSError`, for example:

```ts
import { NotUnixFSError }  from '@helia/unixfs/errors'
```

**Review by achingbrain — CHANGES_REQUESTED** [review]
LGTM, DX nit about throwing generic errors vs named errors.

We've lost the `.code` property so anyone making decisions on thrown errors needs another way to do it, `libp2p`/`helia` now use `.name` - here we're not using anything at all so the dev has to do it based on error messages which can be fragile.

**Comment by 2color** [comment]
> We've lost the .code property so anyone making decisions on thrown errors needs another way to do it, libp2p/helia now use .name - here we're not using anything at all so the dev has to do it based on error messages which can be fragile.

I've adapted to code to either reuse or define custom error types with the `name` property here.

**Comment by 2color** [comment]
It seems that somehow this PR has caused a regression in the Gateway Conformance. 

@SgtPooki Any pointers on debugging this?

**Review by achingbrain — APPROVED** [review]

**Comment by achingbrain** [comment]
You may wish to change the PR title before merging.  `chore: ` will not cause a release but you've added `!` which means a breaking change so maybe you do want a release.  Suggest `deps!:` or `feat!:` or something.

**Comment by achingbrain** [comment]
It doesn't look like the gateway conformance tests have passed for months: https://github.com/ipfs/helia-verified-fetch/actions/workflows/gateway-conformance.yml

**Pull request merged**